src/API/BaseController.php

```diff
@@ -149,6 +149,18 @@ public function atomicMethod(...$args)
      */
     public function getModel($modelNameString = false)
     {
+        // Attempt to load a model using native Phalcon controller API.
+        $controllerClass = "\\PhalconRest\Controllers\\" . $this->getControllerName("singular");
+        if(class_exists($controllerClass)) {
+            $controller = new $controllerClass();
+            $model = $controller->getModel();
+    
+            if($model) {
+                $this->model = $model;
+            }
+        }
+
+        // Backup model loading
         if ($this->model == false) {
```

Comment:
Consider moving this up.  If the model is already defined, we won't need to run the new primary model loading logic you added.